test-utils/state-viewer/src/apply_chain_range.rs

```diff
@@ -0,0 +1,288 @@
+    let end_height = end_height.unwrap_or_else(|| chain_store.head().unwrap().height);
+    let start_height = start_height.unwrap_or_else(|| chain_store.tail().unwrap());
+
+    info!(
+        "Applying chunks in the range {}..={} for shard_id {}",
+        start_height, end_height, shard_id
+    );
+
+    debug!("============================");
+    debug!("Printing results including outcomes of applying receipts");
+
+    let progress_bar = ProgressBar::new(end_height - start_height + 1);
+    progress_bar.set_style(
+        ProgressStyle::default_bar().template("[{elapsed_precise}] {bar:80} {pos:>8}/{len:8}"),
+    );
+    let processed_blocks_cnt = AtomicU64::new(0);
+    // For some reason I can't call `par_iter` directly on the range.
+    (start_height..=end_height).collect::<Vec<u64>>().par_iter().progress_with(progress_bar).for_each(|height| {
+        let height = *height;
+        let mut chain_store = ChainStore::new(store.clone(), genesis.config.genesis_height);
+        let block_hash = if let Ok(block_hash) = chain_store.get_block_hash_by_height(height) {
+            block_hash
+        } else {
+            // Skipping block because it's not available in ChainStore.
+            inc_and_report_progress(&processed_blocks_cnt);
+            return;
+        };
+        let block = chain_store.get_block(&block_hash).unwrap().clone();
+        let shard_uid =
+            runtime_adapter.shard_id_to_uid(shard_id, block.header().epoch_id()).unwrap();
+        assert!(block.chunks().len() > 0);
+        let mut existing_chunk_extra = None;
+        let mut prev_chunk_extra = None;
+        let apply_result = if *block.header().prev_hash() == CryptoHash::default() {
+            info!(target:"state-viewer", "Skipping the genesis block #{}.", height);
+            inc_and_report_progress(&processed_blocks_cnt);
+            return;
+        } else if block.chunks()[shard_id as usize].height_included() == height {
+            let res_existing_chunk_extra = chain_store.get_chunk_extra(&block_hash, &shard_uid);
+            assert!(res_existing_chunk_extra.is_ok(), "Can't get existing chunk extra for block {} at shard {:?}", block_hash, shard_uid);
+            existing_chunk_extra = Some(res_existing_chunk_extra.unwrap().clone());
+            let chunk = chain_store
+                .get_chunk(&block.chunks()[shard_id as usize].chunk_hash())
+                .unwrap()
+                .clone();
+
+            let prev_block = if let Ok(prev_block) =
+            chain_store.get_block(&block.header().prev_hash())
+            {
+                prev_block.clone()
+            } else {
+                info!(target:"state-viewer", "Skipping applying block #{} because the previous block is unavailable and I can't determine the gas_price to use.", height);
+                inc_and_report_progress(&processed_blocks_cnt);
+                return;
+            };
```

Comment:
very weak presonal opinion: `if let {} else {}` most of the time is less readable than the equivalent match 

https://github.com/rust-analyzer/rust-analyzer/blob/master/docs/dev/style.md#if-let